substrate/hash

```diff
@@ -161,8 +161,10 @@ namespace substrate
 		return std::uint16_t((res % m16) + res / m16);
 	}
 
-	template<std::size_t N> std::uint16_t sysv_checksum(std::array<std::uint8_t, N> bytes) noexcept
-		{ return sysv_checksum(bytes.data(), N); }
+	template<typename Sized> inline uint16_t sysv_checksum(const Sized &data) noexcept
+	{
+		return sysv_checksum(reinterpret_cast<const uint8_t *>(substrate::data(data)), substrate::size(data));
```

Comment:
Please add a `// NOLINTNEXTLINE()` for the reinterpret_cast to quiet clang-tidy as we know this is safe (same for `bsd_checksum()`)